### ISSUE TITLE
Regular expressions are not supported

### ISSUE THREAD

**Issue by dan-gph** [badmotorfinger/z#55]
In the Bash version of z, you can match using regular expressions, so for instance `z c.t` would match on `cat` and `cot`. This feature does not appear exist in the PowerShell z, or if it does exist, it doesn't work on my system with z 1.11.13. I simply get a "Not found" message when I try it.

**Comment by badmotorfinger** [comment]
Yes, regular expressions were on my initial feature list when I developed this, however, I got lazy since standard directory lookups have served my needs up until this point.

Happy to accept a pull request for this feature:)
